src/views/docs-view/server.ts

```diff
@@ -304,7 +304,18 @@ export function getStaticGenerationFunctions<
 				// TODO: need to adjust type for sidebarNavDataLevels here
 				sidebarNavDataLevels: sidebarNavDataLevels as $TSFixMe,
 			}
-			if (showVersionSelect) {
+
+			/**
+			 * Determine whether to show the version selector
+			 */
+			const shouldShowVersions = showVersionSelect === true
+			const shouldHideVersions = showVersionSelect === false
+			const hasMeaningfulVersions =
+				versions && (versions.length > 1 || versions[0].version !== 'v0.0.x')
+			if (
+				!shouldHideVersions &&
+				(shouldShowVersions || hasMeaningfulVersions)
+			) {
```

Comment:
thought: It feels kind of off that we have this logic spread across the `getShowVersionSelect` helper as well as here. Do you think it would be feasible to consolidate where this logic lives? If that feels like too large of a refactor, maybe expanding the comment here with some of the context from y[our comment](https://github.com/hashicorp/dev-portal/pull/830#issuecomment-1216729113) could be a good middleground!

Reply:
That feels feasible! I'll take a quick shot at that - will drop `getShowVersionSelect` and move the Packer plugins exception to this location. (Definitely feel free to stop me if I misinterpreted anything! 🙇‍♂️)

Reply:
✅ @BRKalow updated - `getStaticGenerationFunctions` in `docs-view/server` no longer accepts `showVersionSelect`; instead we handle that logic in one place (here in this commented area). Let me know if that looks alright!

Reply:
Looks good! I think we _could_ retain `getShowVersionSelect` and abstract all of the logic into that method, but I don't feel too strongly. I think having it all in one place is the most important! Thanks for following up here 🙇

Reply:
Ah that could make sense! I don't think we'd have access to `versions` in `getShowVersionSelect` where it is in `getRootDocsPathGenerationFunctions` 🤔 Though I guess we could easily move it into `getStaticGenerationFunctions`. I think I'd dig that as well 💯 Will keep that in mind in case I end up touching the versioning display logic again 👍